includes/UserMailer.php

```diff
@@ -98,15 +98,45 @@ class UserMailer {
 	 * @return Status
 	 */
 	protected static function sendWithPear( $mailer, $dest, $headers, $body ) {
-		$mailResult = $mailer->send( $dest, $headers, $body );
 
-		# Based on the result return an error string,
-		if ( PEAR::isError( $mailResult ) ) {
-			wfDebug( "PEAR::Mail failed: " . $mailResult->getMessage() . "\n" );
-			return Status::newFatal( 'pear-mail-error', $mailResult->getMessage() );
-		} else {
-			return Status::newGood();
+		// I wrote this code against my will. They made me do it.
+		//
+		// The send() method called below may:
+		//
+		// a) throw InvalidArgumentException when $headers are bad in some way
+		// b) return an instance of PEAR_Error when $headers are bad in some other way
+		// c) return false when message cannot be sent
+		// d) return true on success
+
+		$success = true; // don't worry, be happy!
+
+		try {
+			$mailResult = $mailer->send( $dest, $headers, $body );
+
+			// handle the b) case:
+			if ( is_a( $mailResult, 'PEAR_Error' ) ) {
```

Comment:
I used `is_a( $mailResult, 'PEAR_Error' )` to be consistent with `Mail2` code and because `PEAR::isError()` was causing fatals (no such class).